eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -361,13 +361,24 @@ jobs:
         set -ex
 
         dockerVolumeArgs="-v $(sourcesPath):/vmr"
-        poisonArg=''
+        sourceOnlyArgs=''
 
-        if [[ '${{ parameters.enablePoison }}' == 'True' ]]; then
-          poisonArg='--poison'
+        if [[ ! -z '${{ parameters.targetOS }}' ]]; then
+          extraBuildProperties="$extraBuildProperties /p:TargetOS=${{ parameters.targetOS }}"
+        fi
+
+        if [[ ! -z '${{ parameters.targetArchitecture }}' ]]; then
+          extraBuildProperties="$extraBuildProperties /p:TargetArchitecture=${{ parameters.targetArchitecture }}"
+        fi
+
+        if [[ '${{ parameters.buildSourceOnly }}' == 'True' ]]; then
+          if [[ '${{ parameters.enablePoison }}' == 'True' ]]; then
+            sourceOnlyArgs='--poison'
+          fi
+          sourceOnlyArgs='$sourceOnlyArgs --source-only /p:SmokeTestsWarnOnSdkContentDiffs=true /p:SmokeTestsExcludeOmniSharpTests=${{ parameters.excludeOmniSharpTests }}'
```

Comment:
Build is failing because variable expansion in Bash doesn't happen in single quotes:

```
$ foo='bar'
$ foo='$foo baz'
$ echo $foo
$foo baz
```

vs

```
$ foo='bar'
$ foo="$foo baz"
$ echo $foo
bar baz
```

Reply:
omg, thanks